### ISSUE TITLE
Specifying cuda version doesn't work

### ISSUE THREAD

**Issue by tgamauf** [fgci-org/ansible-role-cuda#29]
According to the readme it should be possible to specify the cuda version as a role variable. It isn't specified how to define it, though, and none of the ways I tried work.

This probably is just a documentation issue, but how am I supposed to specify the cuda version?

Version: 1.6.6

### Specify the `cuda-10-0` package
```
cuda_packages
    - cuda-10-0
```

Result:
```
googlecompute: TASK [cscfi.cuda : Install CUDA packages (1.5-2GB download, also restarts if cuda_restart_node_on_install is set to True)] ***
googlecompute: failed: [default] (item=cuda=cuda-10-0) => {"ansible_loop_var": "item", "cache_update_time": 1574182675, "cache_updated": false, "changed": false, "item": "cuda=cuda-10-0", "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"      install 'cuda=cuda-10-0'' failed: E: Version 'cuda-10-0' for 'cuda' was not found\n", "rc": 100, "stderr": "E: Version 'cuda-10-0' for 'cuda' was not found\n", "stderr_lines": ["E: Version 'cuda-10-0' for 'cuda' was not found"], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information..."]}
```

### Specify the cuda package version:
```
cuda_packages
    - cuda=10.0
```

Result:
```
googlecompute: TASK [cscfi.cuda : Install CUDA packages (1.5-2GB download, also restarts if cuda_restart_node_on_install is set to True)] ***
googlecompute: failed: [default] (item=cuda=10.0) => {"ansible_loop_var": "item", "cache_update_time": 1574183381, "cache_updated": false, "changed": false, "item": "cuda=10.0", "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"      install 'cuda=10.0'' failed: E: Version '10.0' for 'cuda' was not found\n", "rc": 100, "stderr": "E: Version '10.0' for 'cuda' was not found\n", "stderr_lines": ["E: Version '10.0' for 'cuda' was not found"], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information..."]}
```

**Comment by martbhell** [comment]
Interesting! How about with apt install? 
Which works? 
Maybe the whole packagename like cuda-10-0_10.0.130-1_amd64  or cuda-10-0_10.0.130 if you are on Ubuntu 1804? 

With Ubuntu it seems each ubuntu releaser supports very few cuda versions : https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/

**Comment by tgamauf** [comment]
Sorry, I should have specified - yes, I'm on Ubuntu 18.04!

It was silly of me to just try `10.0` - of course I need to define the full package version. Thanks for pointing that out to me. If I use `cuda=10.0.130-1` installation works like a charm, but the cuda version shown by `nvidia-smi` is still `10.1`. Not exactly sure how and why this happens, as there aren't any cuda 10.1 packages installed ....

**Comment by martbhell** [comment]
Thanks!
I created PR #30 to document how to do that. To be honest I never thought specifying the package like that would actually work. So thank you @tgamauf !

About nvidia-smi I do not follow, at least https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/ has cuda-10.1 packages? 

Also I'm not sure what nvidia-smi checks, probably the kernel modules loaded so if one downgrades cuda-rpm I would maybe still make sure dkms/kernel modules are actually really updated.

(Our openstack's CentOS images also have yum autoupdate on enabled which would update cuda rpm within a day or so. )

**Comment by tgamauf** [comment]
> About nvidia-smi I do not follow, at least https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/ has cuda-10.1 packages?

cuda-10.1 (and now cuda-10.2) packages are available in the repo, but they haven't been installed in the image I created. However, I found a likely cause - it looks like nvidia-smi only shows the maximum supported cuda version, not the currently installed (see the comment by `talonmies` to [this](https://stackoverflow.com/a/55717476/3927228) stackoverflow answer).
I checked the cuda version using nvcc and everything is alright.

Thanks again!
